sql/mysqld.cc

```diff
@@ -8754,6 +8755,16 @@ struct my_option my_long_options[] = {
      "to replicate-do-db.",
      nullptr, nullptr, nullptr, GET_STR, REQUIRED_ARG, 0, 0, 0, nullptr, 0,
      nullptr},
+    {"replica-enable-event", OPT_REPLICA_ENABLE_EVENT,
+     "Tells the replication applier thread to enable the events that match "
+     "the specified wildcard pattern without setting it as SLAVESIDE_DISABLED. "
+     "To specify more than one event, use the directive multiple times, once "
+     "for each event. This will work for cross-database events. Example: "
+     "replica-enable-event=foo%.bar% will enable the events in all "
+     "databases on replica server that start with 'foo' and whose event names "
+     "start with 'bar'.",
+     nullptr, nullptr, nullptr, GET_STR, REQUIRED_ARG, 0, 0, 0, nullptr, 0,
+     nullptr},
```

Comment:
sys_vars.option_basic.test is not necessary?

Reply:
Can't we do SET GLOBAL on this option? any reason why it is 'start-up' only?

Many times, we will not know the events to be replicated at the time of starting server

Reply:
There are some overheads and problems associated with making the values to be updated via SET GLOBAL.

1. It requires a separate parser to add/modify the values.
2. It is not very extensible when someone wants to add more and more patterns to the variable.
3. Requires additional locking for setting the values from multiple threads. With the current approach, there is no need for locking.
4. It becomes a problem on server restart when the variable is not persisted, or it may also need some infrastructure to store the previous value in the system tables.

Because of the above reasons, I found it to be better to be settable through cnf file than through SQL command.
So, there was no need to add a testcase in the `sys_vars` suite and the test `rpl_enable_event_cmd_line_validation.test` covers most of the cases.

Reply:
well, it is extra work, but we can do

1. well, just set is fine. For simplicity, we can take the entire new value provided by the user
2. same as 1 take all or nothing. latest provided by user
3. concurrent SET GLOBAL of an option are already synchronized by mutex
4. SET PERSIST

Reply:
sys_vars suite will mandatorily ask for option_basic.test file isn't it? :thinking: 
It does not depend on the option type

Reply:
I debugged a bit and it seems select * from performance_schema.global_variables doesn't show this variable. That is why sys_vars.all_vars suite doesn't complain.

Can you please check why pfs.global_variables doesn't show this newly added variable. I doesn't show other replicate-* variables as well.. something is happening.

Reply:
Hi @satya-bodapati! Yes. This is expected because of the fact that it is not defined as a subclass of the `sys_var` class to be able to be considered as a system variable.

> Can you please check why pfs.global_variables doesn't show this newly added variable. I doesn't show other replicate-* variables as well.. something is happening.

As replicate-* variables behave in a different way (new value gets on accumulated) when compared to other variables which overrides the previous values when the same variable is specified multiple times.

Upstream introduced new perfschema table for the observability of replicate-* variables. But however, we cannot do the same. So, instead we need to implement a special type of `Sys_var` to be reported under `performance_schema.global_variables` and `SHOW GLOBAL VARIABLES`.

I'm currently working on such an implementation using `Sys_var_charptr_func` and will update the PR if it works.

Reply:
Done. `replica_enable_event` is now a non persistable read only global variable.
```
mysql> show global variables like '%replica_en%';
+----------------------+---------------------------+
| Variable_name        | Value                     |
+----------------------+---------------------------+
| replica_enable_event | "db%.event%","%db%.%ev1%" |
+----------------------+---------------------------+
1 row in set (0.00 sec)

mysql> select * from performance_schema.global_variables where variable_name like '%Replica_en%';
+----------------------+---------------------------+
| VARIABLE_NAME        | VARIABLE_VALUE            |
+----------------------+---------------------------+
| replica_enable_event | "db%.event%","%db%.%ev1%" |
+----------------------+---------------------------+
1 row in set (0.01 sec)

mysql> select @@replica_enable_event;
+---------------------------+
| @@replica_enable_event    |
+---------------------------+
| "db%.event%","%db%.%ev1%" |
+---------------------------+
1 row in set (0.00 sec)

mysql> select @@session.replica_enable_event;
ERROR 1238 (HY000): Variable 'replica_enable_event' is a GLOBAL variable

mysql> set session replica_enable_event = 'db.ev';
ERROR 1238 (HY000): Variable 'replica_enable_event' is a read only variable

mysql> set global replica_enable_event = 'db.ev';
ERROR 1238 (HY000): Variable 'replica_enable_event' is a read only variable
```